run/events_storage/main_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestHelloEventsStorage(t *testing.T) {
+	tests := []struct {
+		subject string
+		want    string
+	}{
+		{subject: "1234", want: "GCS CloudEvent type: 1234\n"},
+	}
+	for _, test := range tests {
+		r, w, _ := os.Pipe()
+		log.SetOutput(w)
+		originalFlags := log.Flags()
+		log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+
```

Comment:
There are no test assertions on log output, so code to redirect logs can be skipped.

Reply:
Without this, we get the date and time in the `string(out)` like `2020/08/10 12:16:43` which causes an error. Is there something we suggest instead?

Reply:
No, you're right. I misread the read below as pulling from the HTTP response instead of the logs. Tyler's comment about using defer to restore the log config is applicable though.

Reply:
Fixed the defer part. It's an issue in other samples though.